### PR TITLE
feat(saml): option to create minimal SAML metadata file

### LINKED ISSUE #6507: Create minimal SAML metadata in application wizard
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

While creating a SAML application, I'm required to provide a metadata file. Not all providers expose this metadata file or it is clear where to find it. ZITADEL does not provide any guidance in the Wizard on what to do when the file is not available.

### Ideal solution

Allow users to provide ENTITYID and ACSURL with text fields and create the minimal metadata file on-the-fly:

```xml
<?xml version="1.0"?>
<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" entityID="${ENTITYID}">
    <md:SPSSODescriptor protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol urn:oasis:names:tc:SAML:1.1:protocol">
        <md:AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="${ACSURL}" index="0"/>
    </md:SPSSODescriptor>
</md:EntityDescriptor>
```

### Minimum solution

Link to a help article or show the minimum snippet so the user to help them create the metadata file.

### Version

_No response_

### Environment

ZITADEL Cloud

### Additional Context

Also discussed in: https://github.com/zitadel/zitadel/discussions/6427

### PR DESCRIPTION
**Pull request by doncicuto**
In #6507 @mffap proposed that:

> While creating a SAML application, I'm required to provide a metadata file. Not all providers expose this metadata file or it is clear where to find it. ZITADEL does not provide any guidance in the Wizard on what to do when the file is not available. [...] Allow users to provide ENTITYID and ACSURL with text fields and create the minimal metadata file on-the-fly [...]

This PR adds a new option when we create a SAML application (either wizard or pro mode) with two input fields for ENTITYID and ACSURL so a minimal metadata file is created on-the-fly. Also we've the same option available when we visit the app's detail.

Here's a video showing this new feature

https://github.com/zitadel/zitadel/assets/30386061/fd74176a-4c01-4738-9a0d-a688d13c49f1

Closes #6507 

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: console/src/app/pages/projects/apps/app-create/app-create.component.html [hunk 1]
```diff
@@ -395,31 +426,61 @@ <h1>{{ 'APP.PAGES.CREATE_DESC_TITLE' | translate }}</h1>
...
+            <div class="cnsl-secondary-text metadata-option">
+              <span>{{ 'APP.SAML.METADATAOPT3' | translate }}</span>
+              <form [formGroup]="samlConfigForm">
+                <cnsl-form-field class="formfield">
+                  <cnsl-label>{{ 'APP.SAML.ENTITYID' | translate }}</cnsl-label>
+                  <input cnslInput formControlName="entityId" placeholder="https://" (change)="changeEntitityIdOrAcsURL()" />
+                </cnsl-form-field>
+                <cnsl-form-field class="formfield">
+                  <cnsl-label>{{ 'APP.SAML.ACSURL' | translate }}</cnsl-label>
+                  <input cnslInput formControlName="acsURL" placeholder="https://" (change)="changeEntitityIdOrAcsURL()" />
```
Comment (peintnermax) [inline console/src/app/pages/projects/apps/app-create/app-create.component.html:471]:
```suggestion
                  <input cnslInput formControlName="acsURL" placeholder="https://" (change)="changeEntityIdOrAcsURL()" />
```

File: console/src/app/pages/projects/apps/app-create/app-create.component.html [hunk 2]
```diff
@@ -395,31 +426,61 @@ <h1>{{ 'APP.PAGES.CREATE_DESC_TITLE' | translate }}</h1>
...
+                </button>
+              </form>
+            </div>
+
+            <div class="cnsl-secondary-text metadata-option">
+              <span>{{ 'APP.SAML.METADATAOPT3' | translate }}</span>
+              <form [formGroup]="samlConfigForm">
+                <cnsl-form-field class="formfield">
+                  <cnsl-label>{{ 'APP.SAML.ENTITYID' | translate }}</cnsl-label>
+                  <input cnslInput formControlName="entityId" placeholder="https://" (change)="changeEntitityIdOrAcsURL()" />
```
Comment (peintnermax) [inline console/src/app/pages/projects/apps/app-create/app-create.component.html:467]:
```suggestion
                  <input cnslInput formControlName="entityId" placeholder="https://" />
```

Comment (peintnermax) [reply console/src/app/pages/projects/apps/app-create/app-create.component.html:467]: please use the `this.samlConfigForm.valueChanges.subscribe` in the ts file to react to changes so we can remove the `changeEntityIdOrAcsURL()`

File: console/src/app/pages/projects/apps/app-create/app-create.component.ts [hunk 3]
```diff
@@ -440,6 +442,22 @@ export class AppCreateComponent implements OnInit, OnDestroy {
...
+    <md:SPSSODescriptor protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol urn:oasis:names:tc:SAML:1.1:protocol">
+        <md:AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="${this.acsURL?.value}" index="0"/>
+    </md:SPSSODescriptor>
+</md:EntityDescriptor>`
+        : '';
+    if (this.samlAppRequest) {
+      const base64 = Buffer.from(minimalMetadata, 'utf-8').toString('base64');
+      this.samlAppRequest.setMetadataXml(base64);
+    }
+  }
```
Comment (peintnermax) [inline console/src/app/pages/projects/apps/app-create/app-create.component.ts:459]: please move this to `this.samlConfigForm.valueChanges.subscribe`. Thx 🙏

**Review by peintnermax — CHANGES_REQUESTED** [review]
I've made some suggestions to react to changes of the new inputs and would slightly change some styling to have descriptions more consistent across the application.
Can you ensure that the suggestions out of the  `app-detail.component` are also applied the same way in `app-create.component`. The rest is very nice work.. thx @doncicuto 💯

**Comment by doncicuto** [comment]
Sure @peintnermax I'll prepare the changes today or tomorrow. Thanks for your review and the suggestions

File: console/src/app/pages/projects/apps/app-detail/app-detail.component.html [hunk 4]
```diff
@@ -176,11 +181,25 @@ <h3>{{ currentRadioItemAuthType.titleI18nKey | translate }}</h3>
...
               />
-              <button mat-stroked-button (click)="$event.preventDefault(); xmlFileInput.click()">
-                {{ 'APP.SAML.XML' | translate }}
+              <button class="upload-button" mat-stroked-button (click)="$event.preventDefault(); xmlFileInput.click()">
+                {{ 'APP.SAML.UPLOAD' | translate }}
               </button>
             </div>
 
+            <div class="metadata-option">
+              <span>{{ 'APP.SAML.METADATAOPT3' | translate }}</span>
```
[the remark this answers is not in the document]
Comment (doncicuto) [reply console/src/app/pages/projects/apps/app-detail/app-detail.component.html:190]: I'll apply the same class to app-create.component.html

**Review by peintnermax — APPROVED** [review]
nice thx a lot @doncicuto 👍

**Comment by doncicuto** [comment]
@eliobischof it seems that the FF test failed... if you want it to, I may try to test manually with latest Firefox in my machine that SMTP test and update the PR if that cypress test can be fixed/improved

**Comment by muhlemmer** [comment]
> @eliobischof it seems that the FF test failed... if you want it to, I may try to test manually with latest Firefox in my machine that SMTP test and update the PR if that cypress test can be fixed/improved

#6811 as follow up. It is not caused by this PR. But feel free if you want to give it a shot :).

**Pull request merged**